interpreter/exec/eval.ml

```diff
@@ -72,10 +72,12 @@ and admin_instr' =
   | Label of int * instr list * code
   | Frame of int * frame * code
   | Handler of int * catch list * code
-  | Handle of (tag_inst * idx) list option * code
+  | Handle of (tag_inst * handler_target) list option * code
+  | Switching of tag_inst * value stack * ref_ * ctxt
   | Suspending of tag_inst * value stack * ctxt
```

Comment:
Since there is a quadratic blow-up involved in the number of rules for context & bubbling instructions, and they should work very similarly, it might be worth merging these two into one (e.g. Suspending with a ref_ option as target).